lib/facility.rb

```diff
@@ -1,14 +1,77 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees
+
+
+  def initialize(args)
+    @name = args[:name]
+    @address = args[:address]
+    @phone = args[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration') 
+      
+      if vehicle.antique?
+        cost = 25
+        plate_type = :antique
+      elsif vehicle.electric_vehicle?
+        cost = 200
+        plate_type = :ev
+      else
+        cost = 100
+        plate_type = :regular
+      end
+
+    vehicle.set_plate_type(plate_type)
+    vehicle.set_registration_date(Date.today)
+    @collected_fees += cost
+    @registered_vehicles << vehicle
+
+    end
```

Comment:
This all works, great job working through the logic of everything this method needed to do. A few notes:
1. Ideally we avoid nested if statements when possible. How could you refactor line 27 and 29 using Boolean Operators `&&` and/or `||` to remove the nested if?
2. This method is doing a lot. Moving forward, let's try to figure out how to get methods to adhere to the Single Responsibility Principle which means every method should do one thing. You could break this method up be creating helper methods that would store pieces of this logic and then call those methods from inside the register_vehicle class. If you're not sure what this could look like, I encourage you to reach out to a mentor or housemate and ask to pair with them to see how you might refactor this.